tests/baselines/reference/exportDefaultNamespace.js

```diff
@@ -0,0 +1,163 @@
+//// [tests/cases/compiler/exportDefaultNamespace.ts] ////
+
+//// [a.ts]
+// https://github.com/Microsoft/TypeScript/issues/3792
+
+/**
+ * namespace A 1 leading comment
+ */
+export default namespace A {
+    export const A1 = 42;
+} // namespace A 1 trailing comment
+
+/**
+ * namespace A 2 leading comment
+ */
+export default namespace A {
+    export function A2() {
+        return 0xC0FFEE;
+    }
+} // namespace A 2 trailing comment
+
+const namespaceAUsage1 = A.A1;
+
+/**
+ * namespace B 1 leading comment
+ */
+export namespace B {
+    export const B1 = 42;
+} // namespace B 1 trailing comment
+
+/**
+ * namespace B 2 leading comment
+ */
+export namespace B {
+    export function B2() {
+        return 0xC0FFEE;
+    }
+} // namespace B 2 trailing comment
+
+const namespaceBUsage1 = B.B2();
+
+namespace ns {
+    /**
+     * namespace C 1 leading comment
+     */
+    export namespace C {
+        export const C1 = 42;
+    } // namespace C 1 trailing comment
+
+    /**
+     * namespace C 2 leading comment
+     */
+    export namespace C {
+        export function C2() {
+            return 0xC0FFEE;
+        }
+    } // namespace C 2 trailing comment
+
+    const namespaceCUsage1 = C.C1;
+}
+
+namespace ns {
+    /**
+     * namespace C 3 leading comment
+     */
+    export namespace C {
+        export class C3 { }
+    } // namespace C 3 trailing comment
+}
+
+namespace ns {
+    const namespaceCUsage2 = C.C1;
+}
+
+
+//// [b.ts]
+import A, { B } from './a';
+
+const namespaceAExternalUsage1 = A.A1;
+const namespaceBExternalUsage1 = B.B2;
+
+//// [a.js]
+"use strict";
+// https://github.com/Microsoft/TypeScript/issues/3792
+exports.__esModule = true;
+/**
+ * namespace A 1 leading comment
+ */
+var A = {};
+exports["default"] = A;
+(function (A) {
+    A.A1 = 42;
+})(A); // namespace A 1 trailing comment
+/**
+ * namespace A 2 leading comment
+ */
+(function (A) {
+    function A2() {
+        return 0xC0FFEE;
+    }
+    A.A2 = A2;
+})(A); // namespace A 2 trailing comment
+var namespaceAUsage1 = A.A1;
+/**
+ * namespace B 1 leading comment
+ */
+var B = {};

```

Comment:
`B` is never exported, despite the fact it is declared as `export enum B` in the test.